### PR TITLE
Update partition statistics in Glue after partition is created

### LINKED ISSUE #7035: Query creating new partition will fail with Glue statistics enabled
```
trino:lo_test> create table p (a bigint, b bigint) with (partitioned_by = ARRAY['b']);
CREATE TABLE
trino:lo_test> insert into p values(10, 10);
INSERT: 1 row

Query 20210225_144420_00002_j72cd, FAILED, 1 node
http://localhost:8080/ui/query.html?20210225_144420_00002_j72cd
Splits: 5 total, 5 done (100.00%)
CPU Time: 0.2s total,     0 rows/s,     0B/s, 25% active
Per Node: 0.0 parallelism,     0 rows/s,     0B/s
Parallelism: 0.0
Peak Memory: 0B
5.72 [0 rows, 0B] [0 rows/s, 0B/s]

Query 20210225_144420_00002_j72cd failed: Partition [10] not found (Service: AWSGlue; Status Code: 400; Error Code: EntityNotFoundException; Request ID: e0ee80c2-2960-4918-9a0e-1554350a9c56; Proxy: null)
io.trino.spi.TrinoException: Partition [10] not found (Service: AWSGlue; Status Code: 400; Error Code: EntityNotFoundException; Request ID: e0ee80c2-2960-4918-9a0e-1554350a9c56; Proxy: null)
        at io.trino.plugin.hive.metastore.glue.DefaultGlueColumnStatisticsProvider.updatePartitionStatistics(DefaultGlueColumnStatisticsProvider.java:291)
        at io.trino.plugin.hive.metastore.glue.converter.GlueInputConverter.convertPartition(GlueInputConverter.java:71)
        at io.trino.plugin.hive.metastore.glue.GlueHiveMetastore.lambda$addPartitions$28(GlueHiveMetastore.java:917)
        at io.trino.plugin.hive.metastore.glue.converter.GlueToTrinoConverter.mappedCopy(GlueToTrinoConverter.java:242)
        at io.trino.plugin.hive.metastore.glue.GlueHiveMetastore.lambda$addPartitions$29(GlueHiveMetastore.java:917)
        at io.trino.plugin.hive.metastore.glue.GlueMetastoreApiStats.call(GlueMetastoreApiStats.java:35)
        at io.trino.plugin.hive.metastore.glue.GlueHiveMetastore.addPartitions(GlueHiveMetastore.java:913)
        at io.trino.plugin.hive.metastore.cache.CachingHiveMetastore.addPartitions(CachingHiveMetastore.java:752)
        at io.trino.plugin.hive.metastore.cache.CachingHiveMetastore.addPartitions(CachingHiveMetastore.java:752)
        at io.trino.plugin.hive.HiveMetastoreClosure.addPartitions(HiveMetastoreClosure.java:230)
        at io.trino.plugin.hive.metastore.SemiTransactionalHiveMetastore$PartitionAdder.execute(SemiTransactionalHiveMetastore.java:3328)
        at io.trino.plugin.hive.metastore.SemiTransactionalHiveMetastore$Committer.executeAddPartitionOperations(SemiTransactionalHiveMetastore.java:1989)
        at io.trino.plugin.hive.metastore.SemiTransactionalHiveMetastore.commitShared(SemiTransactionalHiveMetastore.java:1403)
        at io.trino.plugin.hive.metastore.SemiTransactionalHiveMetastore.commit(SemiTransactionalHiveMetastore.java:1158)
        at io.trino.plugin.hive.HiveMetadata.commit(HiveMetadata.java:2935)
        at io.trino.plugin.hive.HiveConnector.commit(HiveConnector.java:212)
        at io.trino.transaction.InMemoryTransactionManager$TransactionMetadata$ConnectorTransactionMetadata.commit(InMemoryTransactionManager.java:594)
        at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
        at com.google.common.util.concurrent.TrustedListenableFutureTask$TrustedFutureInterruptibleTask.runInterruptibly(TrustedListenableFutureTask.java:125)
        at com.google.common.util.concurrent.InterruptibleTask.run(InterruptibleTask.java:69)
        at com.google.common.util.concurrent.TrustedListenableFutureTask.run(TrustedListenableFutureTask.java:78)
        at io.airlift.concurrent.BoundedExecutor.drainQueue(BoundedExecutor.java:80)
        at io.trino.$gen.Trino_unknown____20210225_144315_2.run(Unknown Source)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
        at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: com.amazonaws.services.glue.model.EntityNotFoundException: Partition [10] not found (Service: AWSGlue; Status Code: 400; Error Code: EntityNotFoundException; Request ID: e0ee80c2-2960-4918-9a0e-1554350a9c56; Proxy: null)
        at com.amazonaws.http.AmazonHttpClient$RequestExecutor.handleErrorResponse(AmazonHttpClient.java:1819)
        at com.amazonaws.http.AmazonHttpClient$RequestExecutor.handleServiceErrorResponse(AmazonHttpClient.java:1403)
        at com.amazonaws.http.AmazonHttpClient$RequestExecutor.executeOneRequest(AmazonHttpClient.java:1372)
        at com.amazonaws.http.AmazonHttpClient$RequestExecutor.executeHelper(AmazonHttpClient.java:1145)
        at com.amazonaws.http.AmazonHttpClient$RequestExecutor.doExecute(AmazonHttpClient.java:802)
        at com.amazonaws.http.AmazonHttpClient$RequestExecutor.executeWithTimer(AmazonHttpClient.java:770)
        at com.amazonaws.http.AmazonHttpClient$RequestExecutor.execute(AmazonHttpClient.java:744)
        at com.amazonaws.http.AmazonHttpClient$RequestExecutor.access$500(AmazonHttpClient.java:704)
        at com.amazonaws.http.AmazonHttpClient$RequestExecutionBuilderImpl.execute(AmazonHttpClient.java:686)
        at com.amazonaws.http.AmazonHttpClient.execute(AmazonHttpClient.java:550)
        at com.amazonaws.http.AmazonHttpClient.execute(AmazonHttpClient.java:530)
        at com.amazonaws.services.glue.AWSGlueClient.doInvoke(AWSGlueClient.java:10483)
        at com.amazonaws.services.glue.AWSGlueClient.invoke(AWSGlueClient.java:10450)
        at com.amazonaws.services.glue.AWSGlueClient.invoke(AWSGlueClient.java:10439)
        at com.amazonaws.services.glue.AWSGlueClient.executeUpdateColumnStatisticsForPartition(AWSGlueClient.java:9413)
        at com.amazonaws.services.glue.AWSGlueClient.updateColumnStatisticsForPartition(AWSGlueClient.java:9380)
        at io.trino.plugin.hive.metastore.glue.DefaultGlueColumnStatisticsProvider.lambda$updatePartitionStatistics$9(DefaultGlueColumnStatisticsProvider.java:263)
        at java.base/java.util.concurrent.CompletableFuture$AsyncRun.run(CompletableFuture.java:1736)
        at com.google.common.util.concurrent.DirectExecutor.execute(DirectExecutor.java:30)
        at java.base/java.util.concurrent.CompletableFuture.asyncRunStage(CompletableFuture.java:1750)
        at java.base/java.util.concurrent.CompletableFuture.runAsync(CompletableFuture.java:1959)
        at io.trino.plugin.hive.metastore.glue.DefaultGlueColumnStatisticsProvider.lambda$updatePartitionStatistics$10(DefaultGlueColumnStatisticsProvider.java:263)
        at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:195)
        at java.base/java.util.AbstractList$RandomAccessSpliterator.forEachRemaining(AbstractList.java:720)
        at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)
        at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
        at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:913)
        at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
        at java.base/java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:578)
        at io.trino.plugin.hive.metastore.glue.DefaultGlueColumnStatisticsProvider.updatePartitionStatistics(DefaultGlueColumnStatisticsProvider.java:269)
        ... 25 more
```

### PR DESCRIPTION
**Pull request by losipiuk**
fixes #7035

### CODE REVIEW

File: plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/glue/GlueHiveMetastore.java [hunk 1]
```diff
@@ -941,6 +942,11 @@ public void addPartitions(HiveIdentity identity, String databaseName, String tab
                     }
                 }
 
+                for (PartitionWithStatistics partition : partitions) {
+                    // TODO(https://github.com/trinodb/trino/issues/7033) make updates in batch
```
Comment (findepi) [inline plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/glue/GlueHiveMetastore.java:946]: https://github.com/trinodb/trino/issues/7033

**Review by findepi — APPROVED** [review]

**Comment by findepi** [comment]
Do we have a test for this already?

(i know we don't run them on CI anyway, but still)

**Comment by findepi** [comment]
or would it require https://github.com/trinodb/trino/issues/5426?

**Comment by losipiuk** [comment]
AC @findepi added some test code

File: plugin/trino-hive/src/test/java/io/trino/plugin/hive/AbstractTestHive.java [hunk 2]
```diff
@@ -878,7 +878,7 @@ protected Transaction newTransaction()
         return new HiveTransaction(transactionManager, (HiveMetadata) metadataFactory.create());
     }
 
-    interface Transaction
+    public interface Transaction
```
Comment (findepi) [inline plugin/trino-hive/src/test/java/io/trino/plugin/hive/AbstractTestHive.java:881]: perhaps `protected`

File: plugin/trino-hive/src/test/java/io/trino/plugin/hive/metastore/glue/TestHiveGlueMetastore.java [hunk 3]
```diff
@@ -676,6 +694,105 @@ public void testGetPartitionsFilterIsNullWithValue()
                 ImmutableList.of(ImmutableList.of(GlueExpressionUtil.NULL_STRING)));
     }
 
+    @Test
+    public void testUpdateStatisticsOnCreate()
```
Comment (findepi) [inline plugin/trino-hive/src/test/java/io/trino/plugin/hive/metastore/glue/TestHiveGlueMetastore.java:698]: The problem was with Glue, but perhaps the test could be generic for all metastore impls.
(and disabled in those which don't support stats)

**Review by findepi — APPROVED** [review]

File: plugin/trino-hive/src/test/java/io/trino/plugin/hive/metastore/glue/TestHiveGlueMetastore.java (hunk 3 above)
Comment (losipiuk) [reply plugin/trino-hive/src/test/java/io/trino/plugin/hive/metastore/glue/TestHiveGlueMetastore.java:698]: Possibly. I would prefer to look into that as a followup.

**Pull request merged**